<?xml version="1.0"?>
<!--XML-->
    <launch>

    <!-- SETUP ARGUMENTS -->
    <!-- Be sure to change this to your cameras serials -->
    <arg name="camera_right_serial" default="31710260"/>
    <arg name="camera_left_serial" default="31710277"/>
    <arg name="camera_namespace" default="stereo"/>
    <arg name="exposure" default="10000"/>
    <arg name="gain" default="100"/>

    <!-- STEREO <MATCHER> -->
    <group ns="$(arg camera_namespace)">
        <arg name="fps" value="1"/>
        <arg name="frame_width" value="2448"/>
        <arg name="frame_height" value="2048"/>

        <arg name="exposure_auto" value="false"/>
        <arg name="gain_auto" value="false"/>

        <arg name="pipeline" value="video/x-raw, format=(string)GRAY8, width=$(arg frame_width), height=$(arg frame_height),framerate=$(arg fps)/1 ! videoconvert --verbose"/>
        
        <!-- Start the GSCAM nodes -->
        <node ns="left" name="$(arg camera_namespace)_node_left" pkg="gscam" type="gscam" output="screen">
            <param name="camera_name" value="left"/>
            <param name="camera_info_url" value="file:///home/i3dr/.ros/camera_info/$(arg camera_namespace)/$(arg camera_namespace)_left.yaml"/>
            <param name="gscam_config" value="tcamsrc serial=$(arg camera_left_serial) ! $(arg pipeline)"/>
            <param name="frame_id" value="/$(arg camera_namespace)_depth_optical_frame"/>
            <!-- param name="image_encoding" value="mono8" /-->
            <param name="sync_sink" value="true"/>
        </node>

        <node ns="right" name="$(arg camera_namespace)_node_right" pkg="gscam" type="gscam" output="screen">
            <param name="camera_name" value="right"/>
            <param name="camera_info_url" value="file:///home/i3dr/.ros/camera_info/$(arg camera_namespace)/$(arg camera_namespace)_right.yaml"/>
            <param name="gscam_config" value="tcamsrc serial=$(arg camera_right_serial) ! $(arg pipeline)"/>
            <param name="frame_id" value="/$(arg camera_namespace)_depth_optical_frame"/>
            <!-- param name="image_encoding" value="mono8" /-->
            <param name="sync_sink" value="true"/>
        </node>

        <!-- Start the tiscamera control node for each camera -->
        <!-- (optional) set inital camera properties using the parameters [Exposure/Gain/Exposure_Auto/Gain_Auto] -->
        <!-- set camera properties using services from this node
             rosservice call /phobos/left/tiscam_set_exposure_auto -->

        <node ns="left" name="i3dr_stereo_camera_left_ctrl" pkg="i3dr_stereo_camera" type="tiscamera_ctrl.py" output="screen">
            <param name="Serial" value="$(arg camera_left_serial)"/>
            <param name="Width" value="$(arg frame_width)"/>
            <param name="Height" value="$(arg frame_height)"/>
            <param name="FPS" value="$(arg fps)"/>
            <param name="Exposure" value="$(arg exposure)"/>
            <param name="Gain" value="$(arg gain)"/>
            <param name="Exposure_Auto" value="$(arg exposure_auto)"/>
            <param name="Gain_Auto" value="$(arg gain_auto)"/>
        </node>

        <node ns="right" name="i3dr_stereo_camera_right_ctrl" pkg="i3dr_stereo_camera" type="tiscamera_ctrl.py" output="screen">
            <param name="Serial" value="$(arg camera_right_serial)"/>
            <param name="Width" value="$(arg frame_width)"/>
            <param name="Height" value="$(arg frame_height)"/>
            <param name="FPS" value="$(arg fps)"/>
            <param name="Exposure" value="$(arg exposure)"/>
            <param name="Gain" value="$(arg gain)"/>
            <param name="Exposure_Auto" value="$(arg exposure_auto)"/>
            <param name="Gain_Auto" value="$(arg gain_auto)"/>
        </node>


    </group>

</launch>
